manifests/init.pp

```diff
@@ -3,10 +3,16 @@
 #
 # @summary Manage the installation and configuration of the puppet_webhook gem
 #
+# @author Vox Pupuli <voxpupuli@groups.io>
+#
 # @example
 #   include puppet_webhook
 #
-class puppet_webhook {
+# @param pkg_provider Which provider to install puppet_webhook into
+#
+class puppet_webhook(
+  Enum['gem', 'puppet_gem'] $pkg_provider,
+) {
   contain 'puppet_webhook::install'
```

Comment:
Should there be chaining from install -> config?

Reply:
You don't need direct chaining anymore, there will be resource relationships setup where the files in `config` will require `Package['puppet_webhook']`. The contain will ensure that both classes are contained within the init class and aware of each other.

I should note that the config class has no code in it yet, the resource relationships will be setup when I start working on those